hathor/p2p/manager.py

```diff
@@ -317,11 +334,29 @@ def update_whitelist(self) -> Deferred[None]:
             settings.WHITELIST_URL.encode(),
             Headers({'User-Agent': ['hathor-core']}),
             None)
+        # Twisted Agent does not have a direct way to configure the HTTP client timeout
+        # only a TCP connection timeout. The callLater below is a manual client timeout
+        # that will cancel the deferred in case it's called
+        timeout_call = self.reactor.callLater(10, d.cancel)
+        d.addBoth(self._update_whitelist_cancel_timeout, timeout_call)
         d.addCallback(readBody)
         d.addErrback(self._update_whitelist_err)
         d.addCallback(self._update_whitelist_cb)
         return d
 
+    def _update_whitelist_cancel_timeout(self, param: Union[Failure, Optional[bytes]],
```

Comment:
Maybe rename it to `_whitelist_update_timeout(...)`?

Reply:
I tried to keep a similar name than the other methods

```
update_whitelist
_update_whitelist_cb
_update_whitelist_err
_update_whitelist_cancel_timeout
```

Reply:
I'd remove the cancel. Maybe just _update_whitelist_timeout?